comp/core/tagger/impl-remote/remote.go

```diff
@@ -0,0 +1,724 @@
+// If possible, avoid using this function, and use the Tag method instead.
+// This function exists in order not to break backward compatibility with rtloader and python
+// integrations using the tagger
+func (t *remoteTagger) LegacyTag(entity string, cardinality types.TagCardinality) ([]string, error) {
+	prefix, id, err := taggercommon.ExtractPrefixAndID(entity)
+	if err != nil {
+		return nil, err
+	}
+
+	entityID := types.NewEntityID(prefix, id)
+	return t.Tag(entityID, cardinality)
+}
+
+// AccumulateTagsFor returns tags for a given entity at the desired cardinality.
+func (t *remoteTagger) AccumulateTagsFor(entityID types.EntityID, cardinality types.TagCardinality, tb tagset.TagsAccumulator) error {
+	tags, err := t.Tag(entityID, cardinality)
+	if err != nil {
+		return err
+	}
+	tb.Append(tags...)
+	return nil
+}
+
+// Standard returns the standard tags for a given entity.
+func (t *remoteTagger) Standard(entityID types.EntityID) ([]string, error) {
+	entity := t.store.getEntity(entityID)
+	if entity == nil {
+		return []string{}, nil
+	}
+
+	return entity.StandardTags, nil
+}
+
+// GetEntity returns the entity corresponding to the specified id and an error
+func (t *remoteTagger) GetEntity(entityID types.EntityID) (*types.Entity, error) {
+	entity := t.store.getEntity(entityID)
+	if entity == nil {
+		return nil, fmt.Errorf("Entity not found for entityID")
+	}
+
+	return entity, nil
+}
+
+// List returns all the entities currently stored by the tagger.
+func (t *remoteTagger) List() types.TaggerListResponse {
+	entities := t.store.listEntities()
+	resp := types.TaggerListResponse{
+		Entities: make(map[string]types.TaggerListEntity),
+	}
+
+	for _, e := range entities {
+		resp.Entities[e.ID.String()] = types.TaggerListEntity{
+			Tags: map[string][]string{
+				remoteSource: e.GetTags(types.HighCardinality),
+			},
+		}
+	}
+
+	return resp
+}
+
+// GetEntityHash returns the hash for the tags associated with the given entity
+// Returns an empty string if the tags lookup fails
+func (t *remoteTagger) GetEntityHash(entityID types.EntityID, cardinality types.TagCardinality) string {
+	tags, err := t.Tag(entityID, cardinality)
+	if err != nil {
+		return ""
+	}
+	return utils.ComputeTagsHash(tags)
+}
+
+func (t *remoteTagger) AgentTags(cardinality types.TagCardinality) ([]string, error) {
+	ctrID, err := metrics.GetProvider(t.wmeta).GetMetaCollector().GetSelfContainerID()
+	if err != nil {
+		return nil, err
+	}
+
+	if ctrID == "" {
+		return nil, nil
+	}
+
+	entityID := types.NewEntityID(types.ContainerID, ctrID)
+	return t.Tag(entityID, cardinality)
+}
+
+func (t *remoteTagger) GlobalTags(cardinality types.TagCardinality) ([]string, error) {
+	return t.Tag(taggercommon.GetGlobalEntityID(), cardinality)
+}
+
+func (t *remoteTagger) SetNewCaptureTagger(tagger.Component) {}
+
+func (t *remoteTagger) ResetCaptureTagger() {}
+
+func (t *remoteTagger) EnrichTags(tb tagset.TagsAccumulator, originInfo taggertypes.OriginInfo) {
```

Comment:
This implementation is the same as the one in the local tagger.
I think we should find a way to define the function just once and reuse it in both places.

If it's not so straightforward, or you think it's better to do it in a separate PR, I think at least we should add a TODO or something in both functions that says that the code is duplicated and that both functions need to be updated if something changes.